test/config/utility.cc

```diff
@@ -636,6 +641,12 @@ ConfigHelper::buildBaseListener(const std::string& name, const std::string& addr
                                 name, address, filter_chains),
                             listener);
   return listener;
+#else
+  UNREFERENCED_PARAMETER(name);
+  UNREFERENCED_PARAMETER(address);
+  UNREFERENCED_PARAMETER(filter_chains);
+  PANIC("YAML support compiled out");
```

Comment:
Nice error message, maybe the others can have this thrown in as well.

e.g. 
https://github.com/envoyproxy/envoy/pull/27634/files#diff-1cf1654ea1233e6793e9a924bbb6ec36beff732bb7ea2c0230d9cc74b87e55f1R83 

https://github.com/envoyproxy/envoy/pull/27634/files#diff-1cf1654ea1233e6793e9a924bbb6ec36beff732bb7ea2c0230d9cc74b87e55f1R189

etc.

Alternatively, what's the thinking about putting this if-def throughout the codebase vs in `loadFromYaml`?